docs/app-deployment.md

```diff
@@ -97,3 +97,20 @@ metadata:
 - [Environment Kustomization](/k8s/prod/common-overlay/kustomization.yaml) also patches environment specific [global values](/k8s/prod/common-overlay/prod-helmrelease.yaml) and  [image automation defaults](/k8s/prod/common-overlay/automated-helmrelease.yaml).
 - Flux image automation is managed through [custom image automation script](/k8s/scripts/container-update.sh)
 - prod-* image updates are updated/committed to base manifest where as the other updates(if configured by teams) are applied to environment specific patches.
+
+## Testing in Local
+
+If you want to find the effective yaml that will get applied to a environment, you can [install kustomize](https://kubernetes-sigs.github.io/kustomize/installation/)  and run build commands.
+
+- To generate the effective yaml generated by your namespace in an env, you can run  
+  
+  `Note : This won't include platform defaults)`
+  ```bash
+  kustomize build --load_restrictor none k8s/<env>/common-overlay/<your-namespace>
+  ```
+- To generate the effective yaml generated including platform default, you can run 
+
+  `Note : This can be very big as it inlcudes all namespaces, you can temporarily remove other namespaces from bases: in k8s/<env>/common-overlay/kustomization.yaml to make it readable)`
```

Comment:
```suggestion
  `Note : This can be very big as it includes all namespaces, you can temporarily remove other namespaces from bases: in k8s/<env>/common-overlay/kustomization.yaml to make it readable)`
```